pkg/connector/tomatrix.go

```diff
@@ -164,6 +169,95 @@ func (l *LinkedInClient) convertFileToMatrix(ctx context.Context, portal *bridge
 	}, err
 }
 
+func (l *LinkedInClient) convertHostURNToMatrix(ctx context.Context, portal *bridgev2.Portal, intent bridgev2.MatrixAPI, hostURNData *linkedingo.HostURNData, textPart *bridgev2.ConvertedMessagePart) (cmp *bridgev2.ConvertedMessagePart, err error) {
+	if hostURNData.Type != "FEED_UPDATE" || hostURNData.HostURN.IsEmpty() {
+		zerolog.Ctx(ctx).Debug().Any("hostUrnData", hostURNData).Msg("Unhandled hostUrnData")
+		return nil, nil
+	}
+	data, err := l.client.GetFeedDashUpdates(ctx, hostURNData.HostURN)
+	if err != nil {
+		return nil, err
+	}
+
+	// urn:li:fsd_update:(urn:li:activity:id,MESSAGING_RESHARE,EMPTY,DEFAULT,false)
+	updateID := hostURNData.HostURN.String()
+	index := strings.Index(updateID, "(")
+	updateID = updateID[index+1:]
+	index = strings.Index(updateID, ",")
+	updateID = updateID[:index]
+
+	url := "https://www.linkedin.com/feed/update/" + updateID
+	linkPreview := event.BeeperLinkPreview{
+		LinkPreview: event.LinkPreview{
+			CanonicalURL: url,
+			Title:        url,
+		},
+		MatchedURL: url,
+	}
+	if data.Actor != nil {
+		linkPreview.Title = data.Actor.Name.Text
+	}
+	if data.Commentary != nil {
+		text, ok := data.Commentary.Text.(string)
+		if !ok {
+			if commentary, ok := data.Commentary.Text.(map[string]any); ok {
+				text, _ = commentary["text"].(string)
+			}
+		}
+		linkPreview.LinkPreview.Description = text
+	}
+	var imageURL string
+	if data.Thumbnail != nil {
+		imageURL = data.Thumbnail.GetLargestArtifactURL()
+	} else if data.Content != nil {
+		var image *linkedingo.Image
+		if data.Content.ArticleComponent != nil && data.Content.ArticleComponent.LargeImage != nil {
+			image = data.Content.ArticleComponent.LargeImage
+		} else if data.Content.ImageComponent != nil && len(data.Content.ImageComponent.Images) > 0 {
+			image = &data.Content.ImageComponent.Images[0]
+		}
+		if image != nil && len(image.Attributes) > 0 && image.Attributes[0].DetailData != nil && image.Attributes[0].DetailData.VectorImage != nil {
+			imageURL = image.Attributes[0].DetailData.VectorImage.GetLargestArtifactURL()
+		}
+	}
+	if imageURL != "" {
+		resp, err := http.Get(imageURL)
```

Comment:
`http.Get` should basically never be used, everything should use http.NewRequestWithContext and the shared http client